apps/frontend/src/database/index.ts

```diff
@@ -1,25 +1,81 @@
-import { addRxPlugin, createRxDatabase, RxDatabase } from 'rxdb';
-import { RxDBDevModePlugin } from 'rxdb/plugins/dev-mode';
-import { getRxStorageDexie } from 'rxdb/plugins/storage-dexie';
-import { RxDBUpdatePlugin } from 'rxdb/plugins/update';
-import { schema } from './schemas';
-import { TodoCollection } from './schemas/todo.schema';
+import { SessionViewer } from "@/types";
+import {
+    RxDatabase,
+    addRxPlugin,
+    createRxDatabase,
+    removeRxDatabase,
+} from "rxdb";
+import { RxDBDevModePlugin } from "rxdb/plugins/dev-mode";
+import { getRxStorageDexie } from "rxdb/plugins/storage-dexie";
+import { RxDBUpdatePlugin } from "rxdb/plugins/update";
+import { schema } from "./schemas";
+import { ProjectCollection } from "./schemas/project.schema";
+import { ViewerCollection } from "./schemas/viewer.schema";
 
 addRxPlugin(RxDBDevModePlugin);
 addRxPlugin(RxDBUpdatePlugin);
 
 export type DBCollections = {
-  todos: TodoCollection;
-}
+    viewer: ViewerCollection;
+    projects: ProjectCollection;
+};
 
 export type DB = RxDatabase<DBCollections>;
 
-const db: DB = await createRxDatabase({
-  name: 'todos-db',
-  ignoreDuplicate: true,
-  storage: getRxStorageDexie(),
-});
+export const databases = new Map<string, DB>();
 
-await db.addCollections(schema);
+const createDb = async (name: string): Promise<DB> => {
+    if (databases.has(name)) {
+        return databases.get(name) as DB;
+    }
 
-export default db;
+    const db: DB = await createRxDatabase({
+        name,
+        multiInstance: true,
+        eventReduce: true,
+        storage: getRxStorageDexie(),
+        ignoreDuplicate: true,
+    });
+
+    await db.addCollections(schema);
+    databases.set(name, db);
+    return db;
+};
+
+export const createOrgDbs = async (viewer: SessionViewer) => {
+    viewer.organizations?.edges.forEach(async (edge) => {
```

Comment:
@AfeezGL Is this still creating all databases at once? Remember it should only create the database needed for the active workspace.